db/db_impl_readonly.cc

```diff
@@ -137,13 +137,25 @@ Status DB::OpenForReadOnly(const Options& options, const std::string& dbname,
   std::vector<ColumnFamilyDescriptor> column_families;
   column_families.push_back(
       ColumnFamilyDescriptor(kDefaultColumnFamilyName, cf_options));
+  if (db_options.stats_persist_period_sec != 0) {
+    column_families.push_back(
+        ColumnFamilyDescriptor(kPersistentStatsColumnFamilyName, cf_options));
+  }
   std::vector<ColumnFamilyHandle*> handles;
 
   s = DB::OpenForReadOnly(db_options, dbname, column_families, &handles, dbptr);
   if (s.ok()) {
-    assert(handles.size() == 1);
+    if (db_options.stats_persist_period_sec != 0) {
+      assert(handles.size() == 2);
```

Comment:
The interface assumes that the stat CF is a normal CF that users should access using Get() API. Though this may not be a bad idea, this heavily violate backward compatibility. We should think twice about changes breaking backward compatibility.

Reply:
I think the stat CF is very similar to a normal CF. Do you mean we should not expose it via `handles`? @siying

Reply:
I'd prefer an API that returns timeseries stats in a structured way and hides the separate CF. Exposing the raw key-values, where keys are encoded in a specific way, feels a bit low-level.

Reply:
If we return time series stats in a structured way while the underlying storage is kv storage, I think we are somewhat trying to define and implement a 'data model' on top of persistent stats storage. My suggestion would be keeping it simple and clear. This also reminds me of MySQL's `performance_schema`. The data model of `performance_schema` is SQL because MySQL supports SQL. If RocksDB supports only kv storage, then what kind of data model should we support in this case? The answer to this question also largely depends on what kind of queries we would like RocksDB to be able to answer about performance stats. I think supporting the data retrieval within a given time range should be supported. Besides that, do we want to support aggregation? I don't have a clear answer to this, just post the question so that we can take it into account when we discuss.